src/app/src/sensorUtils.js

```diff
@@ -148,24 +149,39 @@ function inRangeInclusive(x, lowerLimit, upperLimit) {
     return lowerLimit <= x && x <= upperLimit;
 }
 
+function checkIfVariableNearEdgeOfHealthyRange(
+    variableValue,
+    lower,
+    upper,
+    variable
+) {
+    // A variable is in fair condition if it is within 10%
+    // of the upper or lower end of the healthy range,
+    // except for turbidity, which is only considered fair
+    // if it is within 10% of the upper end.
+    const isVariableNearEdgeOfHealthyRange =
+        (variable !== TURBIDITY &&
+            inRangeInclusive(
+                variableValue,
+                lower,
+                (upper - lower) / 10 + lower
+            )) ||
+        inRangeInclusive(variableValue, upper - (upper - lower) / 10, upper);
```

Comment:
I think this works, but it's taken me a while to reason through. Consider improving the readability.
```
    const isInUpperTenth = v => inRangeInclusive(v, upper - (upper - lower) / 10, upper);
    const isInLowerTenth = v => inRangeInclusive(
        v,
        lower,
        (upper - lower) / 10 + lower
    );
    const isVariableNearEdgeOfHealthyRange =
        variable === TURBIDITY
        ? isInUpperTenth(variableValue)
        : isInUpperTenth(variableValue) || isInLowerTenth(variableValue);

    return isVariableNearEdgeOfHealthyRange;
```

Reply:
Thanks! Nice improvement. Updated.